src/java_bytecode/java_bytecode_convert_method_class.h

```diff
@@ -32,15 +31,15 @@ class java_bytecode_convert_methodt:public messaget
 {
 public:
   java_bytecode_convert_methodt(
-    symbol_tablet &_symbol_table,
+    symbol_table_baset &symbol_table,
     message_handlert &_message_handler,
     size_t _max_array_length,
-    safe_pointer<ci_lazy_methods_neededt> _lazy_methods,
+    optionalt<ci_lazy_methods_neededt> needed_lazy_methods,
     java_string_library_preprocesst &_string_preprocess):
     messaget(_message_handler),
-    symbol_table(_symbol_table),
+    symbol_table(symbol_table),
     max_array_length(_max_array_length),
-    lazy_methods(_lazy_methods),
+    needed_lazy_methods(std::move(needed_lazy_methods)),
```

Comment:
`needed_lazy_methods` is itself a pair of references, so this actually doesn't change semantics, but appears to. Consider straightforward pass-by-value?

Reply:
It halves the amount of memory copied from two pointers to one!